### PR TITLE
Enforce the absence of static initializers

### PR DESCRIPTION
**Pull request by litherum**
#### 3d46683588f3186fe423ad1effa5f8392d9af1e0
<pre>
Enforce the absence of static initializers
<a href="https://bugs.webkit.org/show_bug.cgi?id=242428">https://bugs.webkit.org/show_bug.cgi?id=242428</a>

Reviewed by Yusuke Suzuki.

Add -Xlinker -no_inits to the LDFLAGS to make sure that static initalizers cause build errors.

* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/Configurations/adattributiond.xcconfig:
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:

Canonical link: <a href="https://commits.webkit.org/252255@main">https://commits.webkit.org/252255@main</a>
</pre>

### CODE REVIEW

**Comment by litherum** [comment]
Uh oh.

```
ld: static initializer found in '/Users/mmaxfield/Build/Products/WebCore.build/Debug/WebCore.build/Objects-normal/arm64e/UnifiedSource88.o' file '/Users/mmaxfield/Build/Products/WebCore.build/Debug/WebCore.build/Objects-normal/arm64e/UnifiedSource88.o'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by litherum** [comment]
Looks like `DEFINE_DEBUG_ONLY_GLOBAL` violates this.

**Comment by litherum** [comment]
Looks like the build completes successfully in release mode, but not debug mode. I should make this linker flag only present on non-debug builds.

File: Source/bmalloc/Configurations/Base.xcconfig [hunk 1]
```diff
@@ -168,3 +168,5 @@ WK_USER_LTO_MODE_thin = YES_THIN;
 WK_USER_LTO_MODE_none = NO;
 WK_USER_LTO_MODE_ = $(WK_DEFAULT_LTO_MODE);
 WK_DEFAULT_LTO_MODE = $(WK_USER_LTO_MODE_thin);
+
+OTHER_LDFLAGS = $(inherited) -Xlinker -no_inits;
```
Comment (Constellation) [inline Source/bmalloc/Configurations/Base.xcconfig]: Do we need this? libbmalloc is built as static library, so we actually link things at JavaScriptCore.framework layer.

File: Source/WTF/Configurations/Base.xcconfig [hunk 2]
```diff
@@ -181,3 +181,5 @@ WK_LIBRARY_INSTALL_PATH_OVERRIDE_YES = /usr/local/lib/safari-sdk;
 WTF_INSTALL_PATH_PREFIX = $(WTF_INSTALL_PATH_PREFIX_DEPLOYMENT_$(DEPLOYMENT_LOCATION));
 WTF_INSTALL_PATH_PREFIX_DEPLOYMENT_YES = $(WTF_INSTALL_PATH_PREFIX_DEPLOYMENT_YES_USE_ALTERNATE_$(WK_USE_ALTERNATE_FRAMEWORKS_DIR));
 WTF_INSTALL_PATH_PREFIX_DEPLOYMENT_YES_USE_ALTERNATE_YES = $(WK_ALTERNATE_FRAMEWORKS_DIR)/;
+
+OTHER_LDFLAGS = $(inherited) -Xlinker -no_inits;
```
Comment (Constellation) [inline Source/WTF/Configurations/Base.xcconfig]: Do we need this? libWTF is built as static library, so we actually link things at JavaScriptCore.framework layer.

**Review by Constellation — APPROVED** [review]
r=me with comments.

**Pull request merged**